.github/workflows/compile-macos-intel.yaml

```diff
@@ -0,0 +1,35 @@
+name: MacOS(ARM64)
+
+on: [pull_request]
+
+jobs:
+  compile-MacOS-ARM64:
+    runs-on: macos-latest
+    steps:
+    - name: Check Platform
```

Comment:
IMHO, this platform check in unnecessary. GHA's macOS will not go back to x86_64.

Reply:
Thanks for the context. That's good to know.

But I have a trust issue with 3rd-party packages, it may never be an issue but double-checking gives me peace of mind 😄

Reply:
If you insist on having a check that is purely defensive in nature, then you need to put a comment there explaining it (because others will have exactly the same question).